File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/result/WorkAreaResult.java
package com.dozycoffee.wms.warehouse.application.port.in.result;

import com.dozycoffee.wms.warehouse.domain.enumeration.AreaCode;
import com.dozycoffee.wms.warehouse.domain.enumeration.AvailabilityStatus;
import com.dozycoffee.wms.warehouse.domain.model.WorkArea;

public record WorkAreaResult(
        Long workAreaId,
        Long warehouseId,
        AreaCode areaCode,
        String areaName,
        int maxCapacity,
        int usedCapacity,
        AvailabilityStatus workAreaStatus
) {

    public static WorkAreaResult from(WorkArea workArea) {
        return new WorkAreaResult(
                workArea.getWorkAreaId(),
                workArea.getWarehouseId(),
                workArea.getAreaCode(),
                workArea.getAreaName(),
                workArea.getCapacity().value(),
                workArea.getUsedCapacity(),
                workArea.getWorkAreaStatus()
        );
    }
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/result/WorkAreaResult.kt
package com.dozycoffee.wms.warehouse.application.port.`in`.result

import com.dozycoffee.wms.warehouse.domain.enumeration.AreaCode
import com.dozycoffee.wms.warehouse.domain.enumeration.AvailabilityStatus
import com.dozycoffee.wms.warehouse.domain.model.WorkArea

data class WorkAreaResult(
    val workAreaId: Long,
    val warehouseId: Long,
    val areaCode: AreaCode,
    val areaName: String,
    val maxCapacity: Int,
    val usedCapacity: Int,
    val workAreaStatus: AvailabilityStatus
) {
    companion object {
        fun from(workArea: WorkArea): WorkAreaResult {
            return WorkAreaResult(
                requireNotNull(workArea.workAreaId),
                workArea.warehouseId,
                workArea.areaCode,
                workArea.areaName,
                workArea.capacity.value,
                workArea.usedCapacity,
                workArea.workAreaStatus
            )
        }
    }
}
